src/system/zephyr/system.c

```diff
@@ -96,7 +96,12 @@ void z_task_free(z_task_t **task) {
 /*------------------ Mutex ------------------*/
 int8_t z_mutex_init(z_mutex_t *m) { return pthread_mutex_init(m, 0); }
 
-int8_t z_mutex_free(z_mutex_t *m) { return pthread_mutex_destroy(m); }
+int8_t z_mutex_free(z_mutex_t *m) {
+    if (*m == NULL) {
```

Comment:
Do you really mean `if(m == NULL)` ?

Reply:
No in this case the pointer is fine as it's the address of a struct field so only the value should cause problem.

Although I could have add a (m==NULL) too.